backend/internal/handlers/webhooks/webhooks.go

```diff
@@ -41,3 +47,56 @@ func (s *WebHookService) PRThread(c *fiber.Ctx) error {
 	println("PR thread webhook event")
 	return c.SendStatus(200)
 }
+
+func (s *WebHookService) PushEvent(c *fiber.Ctx) error {
+	// Extract the 'payload' form value
+	payload := c.FormValue("payload")
+	if payload == "" {
+		return errs.BadRequest(errors.New("missing payload"))
+	}
+
+	// Unmarshal the JSON payload into the PushEvent struct
+	var pushEvent github.PushEvent
+	err := json.Unmarshal([]byte(payload), &pushEvent)
+	if err != nil {
+		return errs.BadRequest(errors.New("invalid JSON payload"))
+	}
+
+	// Check if this is first commit in a repository
+	isInitialCommit, err := s.isInitialCommit(pushEvent)
+	if err != nil {
+		return err
+	}
+
+	// If app bot triggered the initial commit, initialize the base repository
+	if isInitialCommit && pushEvent.Pusher != nil && *pushEvent.Pusher.Name == "khoury-classroom[bot]" {
+		err = s.baseRepoInitialization(c, pushEvent)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *WebHookService) baseRepoInitialization(c *fiber.Ctx, pushEvent github.PushEvent) error {
+	if pushEvent.Repo == nil || pushEvent.Repo.Organization == nil || pushEvent.Repo.Name == nil || pushEvent.Repo.MasterBranch == nil {
+		return errs.BadRequest(errors.New("invalid repository data"))
+	}
+
+	// Create development branch
+	_, err := s.githubappclient.CreateBranch(c.Context(), *pushEvent.Repo.Organization,
+		*pushEvent.Repo.Name,
+		*pushEvent.Repo.MasterBranch,
+		"dev")
+	if err != nil {
+		return errs.InternalServerError()
+	}
+
+	return c.SendStatus(200)
+}
+
+func (s *WebHookService) isInitialCommit(pushEvent github.PushEvent) (bool, error) {
```

Comment:
remove the (s *WebHookService) as this is just a local helper function and not actually a webhook service function